### DISCUSSION TITLE
Podman build fails to build Node20 images

### DISCUSSION THREAD

**Discussion by jshah04** [containers/podman#20682] [General]
### Issue Description

I am using podman build in my CI jobs to build our images and publish them. However, the podman build fails on npm install and other npm steps for images released after node18.10.0-alpine3.15. I have tried multiple newer node18 images and most of node20 images. Below I have attached the dockerfile where I was seeing the issue. 

**Variables used in the file -** 
VARIANT: node20
NODE_IMAGE: "node:20.9.0-alpine3.18"
YARN_VERSION: "v1.22.19"
KANIKO_IMAGE: "gcr.io/kaniko-project/executor:v1.8.1-debug"
JDK_IMAGE: "amazoncorretto:8-alpine3.15-jdk"
VAULT_VERSION: "1.10.1"

**Dockerfile -** 
ARG NODE_IMAGE
ARG KANIKO_IMAGE
ARG JDK_IMAGE

FROM $NODE_IMAGE as nodeImage

FROM $KANIKO_IMAGE as kanikoImage

FROM $JDK_IMAGE as base

ENV DOCKER_CONFIG /kaniko/.docker/
ENV PATH $PATH:/kaniko

RUN mkdir -p /kaniko/.docker && \
  apk -uv add --no-cache \
  groff \
  jq \
  aws-cli \
  unzip \
  curl \
  bash \
  make \
  git \
  g++ \
  python2 \
  openssh \
  autoconf \  
  build-base \
  apk-tools \
  zip

ARG VAULT_VERSION
RUN wget --quiet  https://releases.hashicorp.com/vault/${VAULT_VERSION}/vault_${VAULT_VERSION}_linux_amd64.zip && \
  unzip vault_${VAULT_VERSION}_linux_amd64.zip -d /usr/bin 

ARG YARN_VERSION
COPY --from=nodeImage /usr/local/bin/ /usr/local/bin/
COPY --from=nodeImage /usr/local/lib/node_modules /usr/local/lib/node_modules
COPY --from=nodeImage /opt/yarn-$YARN_VERSION /opt/yarn-$YARN_VERSION

COPY --from=kanikoImage /kaniko /kaniko

WORKDIR /workspace

RUN rm -rf ~/.npm/_cacache ~/.npmrc .npmrc \
  && npm cache verify \
  && npm config list -l \ 
  && npm i -g typescript ts-node jest-junit xml

**Job Logs where failure occurs -** 
[3/3] STEP 7/33: ARG YARN_VERSION
--> 69019ad4f54
[3/3] STEP 8/33: COPY --from=nodeImage /usr/local/bin/ /usr/local/bin/
--> 2c9a6d9db1d
[3/3] STEP 9/33: COPY --from=nodeImage /usr/local/lib/node_modules /usr/local/lib/node_modules
--> 2b7dd22d85e
[3/3] STEP 10/33: COPY --from=nodeImage /opt/yarn-$YARN_VERSION /opt/yarn-$YARN_VERSION
--> 96b10fcf557
[3/3] STEP 11/33: COPY --from=kanikoImage /kaniko /kaniko
--> b29ea507c7b
[3/3] STEP 12/33: WORKDIR /workspace
--> 8e1910e0a67
[3/3] STEP 13/33: RUN rm -rf ~/.npm/_cacache ~/.npmrc .npmrc 
--> c3f94e03d18
[3/3] STEP 14/33: RUN npm cache verify 
Assertion failed: fd > STDERR_FILENO (../deps/uv/src/unix/core.c: uv__close: 633)
subprocess exited on aborted
subprocess exited with status 1
Error: error building at STEP "RUN npm cache verify": exit status 1

Please keep in mind that removing npm cache verify just makes it fail on a different npm step.

### Steps to reproduce the issue

Steps to reproduce the issue
1. Build dockerfile with variables specified using Podman

### Describe the results you received

Image fails to build with the error linked above.

### Describe the results you expected

I expected my image to built successfully as it has for images up until node18.10.0-alpine-3.15

### podman info output

```yaml
time="2023-11-08T16:20:38Z" level=warning msg="\"/\" is not a shared mount, this could cause issues or missing mounts with rootless containers"
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: Unknown
    path: /usr/bin/conmon
    version: 'conmon version 2.0.30, commit: 7ec8c3be38789b5e932d932f3e478142a8c263ef'
  cpus: 4
  distribution:
    distribution: alpine
    version: 3.15.8
  eventLogger: file
  hostname: runner-pxqsnb-5-project-29976493-concurrent-0-duw6mmpl
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 0
      size: 1
    - container_id: 1
      host_id: 1
      size: 4294967294
    uidmap:
    - container_id: 0
      host_id: 0
      size: 1
    - container_id: 1
      host_id: 1
      size: 4294967294
  kernel: 5.10.197-186.748.amzn2.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 12712984576
  memTotal: 16454746112
  ociRuntime:
    name: crun
    package: Unknown
    path: /usr/bin/crun
    version: |-
      crun version 1.3
      commit: 4f6c8e0583c679bfee6a899c05ac6b916022561b
      spec: 1.0.0
      +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    path: /run/user/0/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /etc/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: Unknown
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 0
  swapTotal: 0
  uptime: 1h 54m 0.59s (Approximately 0.04 days)
plugins:
  log:
  - k8s-file
  - none
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /root/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: vfs
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphStatus: {}
  imageStore:
    number: 0
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.4.7
  Built: 1652332308
  BuiltTime: Thu May 12 05:11:48 2022
  GitCommit: 66aa35dd4350231801ee4c351b7066c63d860693
  GoVersion: go1.17.10
  OsArch: linux/amd64
  Version: 3.4.7
```

### Podman in a container

Yes

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by Luap99** [comment]
> Version: 3.4.7

Note you are using a very old podman version, I highly suggest you try to update to the latest version and try again.

**Comment by jshah04** [comment]
I tried to fetch the latest podman version using `RUN apk add podman=4.7.1-r0` as found in the packages for alpine linux but I am seeing the following error - 
` STEP 27/33: RUN apk add podman=4.7.1-r0
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
fetch https://apk.corretto.aws/x86_64/APKINDEX.tar.gz
ERROR: unable to select packages:
  podman-3.4.7-r1:
    breaks: world[podman=4.7.1-r0]
subprocess exited with status 1
subprocess exited with status 1
Error: error building at STEP "RUN apk add podman=4.7.1-r0": exit status 1`
Please advice.

**Comment by rhatdan** [comment]
This feels a lot more like a discussion then a podman issue.

**Reply by jshah04** [reply]
Thank you. Do you have any suggestions on why this might be happening?
https://pkgs.alpinelinux.org/packages?name=podman&branch=edge&repo=&arch=&maintainer=
I see 4.7.2 here.
